test/cypress/integration/widgets/deployment_actions_spec.js

```diff
@@ -53,4 +53,109 @@ describe('Deployment Action Buttons widget', () => {
             cy.get('.modal').should('not.exist');
         });
     });
+
+    describe('should allow to manage deployment labels', () => {
+        function typeLabelKey(key) {
+            cy.get('div[name=labelKey] > input').clear().type(key);
+        }
+        function typeLabelValue(value) {
+            cy.get('div[name=labelValue] > input').clear().type(value);
+        }
+        function addLabel(key, value) {
+            typeLabelKey(key);
+            typeLabelValue(value);
+            cy.get('button .add').click();
+        }
+
+        before(() => {
+            cy.setLabels(deploymentName, [{ existing_key: 'existing_value' }]);
+            cy.setDeploymentContext(deploymentName);
+            cy.interceptSp('GET', `/deployments/${deploymentName}?_include=labels`).as('fetchLabels');
+            cy.get('button.deploymentActionsButton').click();
+            cy.get('.popupMenu > .menu').contains('Manage Labels').click();
+            cy.get('.modal').within(() => {
+                cy.wait('@fetchLabels');
+                cy.get('form.loading').should('not.exist');
+                cy.get('.selection').click();
+            });
+        });
+
+        beforeEach(() => {
+            // NOTE: Clicking at the header to close opened dropdowns / popups
+            cy.get('.modal .tags').click();
```

Comment:
If we use classes anyway, let's at least make the classname a bit less confusing IMO :smile: 

```suggestion
            cy.get('.modal .header').click();
```

I did not verify that it works :slightly_smiling_face: but it's a bit more readable to me

Reply:
Done.